flocker/node/script.py

```diff
@@ -312,7 +312,7 @@ def get_service(self, reactor, options):
         configuration = get_configuration(options)
         host = configuration['control-service']['hostname']
         port = configuration['control-service']['port']
-        ip = self.get_external_ip(host, port)
+        ip = configuration.get('hostname', self.get_external_ip(host, port))
```

Comment:
I suggest only running get_external_ip if configuration doesn't have an entry, given it adds noise to control service logs.